tree/30_generic_methods/user_fullname.cf

```diff
@@ -0,0 +1,61 @@
+#####################################################################################
+# Copyright 2017 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name User fullname
+# @description Define the fullname of the user, user must already exists.
+# @documentation This method does not create the user.
+#
+# @parameter login User's login
+# @parameter fullname User's fullname
+# 
+# @class_prefix user_fullname
+# @class_parameter login
+
+bundle agent user_fullname(login, fullname)
+{
+  vars:
+      "old_class_prefix"  string => canonify("user_fullname_${login}");
+      "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+      "class_prefix"      string => canonify(join("_", "promisers"));
+      "args"               slist => { "${login}", "${fullname}" };
+      
+  classes:
+      "user_exists"     expression => userexists("${login}");
+      "args_defined"    expression => "user_exists";
```

Comment:
why creating args_defined ? and not simply using user_exists ?

Reply:
Not useful, but I wrote all this at the same time and chose to use same syntax everywhere.